test/test.c

```diff
@@ -51,58 +66,86 @@
-
-	CHECKJSON("true");
-	CHECKJSON("false");
-
-	CHECKJSON("null");
-
-	CHECKJSON("[]");
-	CHECKJSON("[1]");
-	CHECKJSON("[\"aaaaaaaaaaaaaaaaaaaaaaaaaaa\",42]");
-	CHECKJSON("[1,2,\"hoi\",\"\\\\\"]");
-	CHECKJSONERR("[1,2,\"hoi\",\"\\\"][]");
-	CHECKJSON("[[1,2],3]");
-	CHECKJSON("[[[[[[[[1],1],1],1],[[1],1],[[[[1],1],1],1],1],1],1],1]");
-	CHECKJSONERR("[[[[[[[[1],1],1],1],[[1],1],[[[[[1],1],1],1],1],1],1],1]");
-	CHECKJSON("[[[[[1],2],[3],4],[[5],6],[7],8],[[[9],0],[1],2],[[3],4],[5],6]");
-
-	CHECKJSON("{}");
-	CHECKJSON("{\"a\":[1,2],\"kaas\":null}");
-	CHECKJSONERR("{\"obj\":{\"1\":2.\"3\":4},\"x\":\"y\"}");
-	CHECKJSON("{\"obj\":{\"1\":2,\"3\":4},\"x\":\"y\"}");
-	CHECKJSONERR("{\"obj\":{\"1\":2.\"3\":4},\"x\":y}");
-
-
-	CHECKBIDIREQSAME("null");
-	CHECKBIDIREQSAME("false");
-	CHECKBIDIREQSAME("[1,2,3]");
-	CHECKBIDIREQSAME("\"iets\"");
-	CHECKBIDIRNEQ("true","false");
-	CHECKBIDIREQ("{\"a\":\"\\u003c\t\n\fkaas\\\"\",\"iets\":[]}",
-	             "{\"a\":\"<\\t\\n\\fkaas\\\"\",\"iets\":[]}");
-
-
-	OK();
+	int ranTotal = 0,
+	    passedTotal = 0;
+
+	SECTION("empty", {
+		CHECKJSONERR("");
+		CHECKJSONERR("    ");
+		CHECKJSONERR("\t\t");
+	});
+
+	SECTION("numbers", {
+		CHECKJSON("123");
+		CHECKJSON("21.4e-3");
+		CHECKJSONERR("123a");
+	});
+
+	SECTION("strings", {
+		CHECKJSON("\"kaas\"");
+		CHECKJSON("\"kaa\\u003cs\\\"\"");
+		CHECKJSONERR("\"");
+		CHECKJSON("\"\\uacef\"");
+		CHECKJSONERR("\"\\uaceg\"");
+		CHECKJSONERR("\"dingen\\\"");
+
+		CHECKBIDIREQSAME("\"iets\"");
+	});
+
+	SECTION("identifiers", {
+		CHECKJSON("true");
+		CHECKJSON("false");
+
+		CHECKJSON("null");
+
+		CHECKBIDIREQSAME("null");
+		CHECKBIDIREQSAME("false");
+		CHECKBIDIRNEQ("true","false");
+	});
+
+	SECTION("arrays", {
+		CHECKJSON("[]");
+		CHECKJSON("[1]");
+		CHECKJSON("[\"aaaaaaaaaaaaaaaaaaaaaaaaaaa\",42]");
+		CHECKJSON("[1,2,\"hoi\",\"\\\\\"]");
+		CHECKJSONERR("[1,2,\"hoi\",\"\\\"][]");
+		CHECKJSON("[[1,2],3]");
+		CHECKJSON("[[[[[[[[1],1],1],1],[[1],1],[[[[1],1],1],1],1],1],1],1]");
+		CHECKJSONERR("[[[[[[[[1],1],1],1],[[1],1],[[[[[1],1],1],1],1],1],1],1]");
+		CHECKJSON("[[[[[1],2],[3],4],[[5],6],[7],8],[[[9],0],[1],2],[[3],4],[5],6]");
+		CHECKJSON("[\"hello\"]");
+		CHECKJSON("[ \"hello\" ]");
+		CHECKJSON("[ \"hello\", \"world\" ]");
+		CHECKJSON("[ 1, 2 ]");
+
+		CHECKBIDIREQSAME("[1,2,3]");
+	});
+
+	SECTION("objects", {
+		CHECKJSON("{}");
+		CHECKJSON("{\"a\":[1,2],\"kaas\":null}");
+		CHECKJSONERR("{\"obj\":{\"1\":2.\"3\":4},\"x\":\"y\"}");
+		CHECKJSON("{\"obj\":{\"1\":2,\"3\":4},\"x\":\"y\"}");
+		CHECKJSONERR("{\"obj\":{\"1\":2.\"3\":4},\"x\":y}");
+		CHECKJSON("{ \"kaas\": [ \"is\", \"lekker\" ] }");
+
+		CHECKBIDIREQ("{\"a\":\"\\u003c\t\n\fkaas\\\"\",\"iets\":[]}",
+		             "{\"a\":\"<\\t\\n\\fkaas\\\"\",\"iets\":[]}");
+	});
+
+	bool successful = passedTotal == ranTotal;
+	printf(color("%d failed out of %d\n", successful ? "green" : "red"), ranTotal - passedTotal, ranTotal);
+	return successful ? 0 : 1;
```

Comment:
@tomsmeding It also returns a correct exit code, so we can also use travis-ci for this repo.